### PR TITLE
React native async storage

### PR DESCRIPTION
**Pull request by mounir95**

### CODE REVIEW

File: src/excels/sorttingfilters/sorts/SortByDate.tsx [hunk 1]
```diff
@@ -1,33 +1,44 @@
...
 
   const setDateSort = (event: React.ChangeEvent) => {
-    const datesorting = event.toString() === 'desc' ? 'asc' : 'desc';
-    const newobjectarray = orderBy(
-      getGlobalObjectStore().arrayofobjects.get(),
-      (obj: TPrObject) => obj.Mydate,
-      [datesorting],
-    )
+    let newobjectarray;
+    if (event.toString() === 'تنازلي' || event.toString() === 'desc') {
```
Comment (AnthonyHaykal) [inline src/excels/sorttingfilters/sorts/SortByDate.tsx]: you can have one check using getLanguageStore.get no?

File: src/stores/LanguageStore.tsx [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    },
+    asc: {
+      AR: 'تصاعدي',
+      ENG: 'asc',
+    },
+    arrayofsort: {
+      AR: ['تصاعدي', 'تنازلي'],
+      ENG: ['asc', 'desc'],
+    },
+    excelcol: {
```
Comment (AnthonyHaykal) [inline src/stores/LanguageStore.tsx:135]: we don't usually use arrays, but its okay for now

File: package.json [hunk 3]
```diff
@@ -26,6 +27,7 @@
     "react-native": "0.67.2",
     "react-native-cli": "^2.0.1",
     "react-native-picker-select": "^8.0.4",
+    "react-native-popup-dialog": "^0.18.3",
```
Comment (AnthonyHaykal) [inline package.json:30]: do we need this?

File: package.json [hunk 4]
```diff
@@ -40,6 +42,7 @@
     "@types/lodash": "^4.14.178",
     "@types/moment": "^2.13.0",
     "@types/react-native": "^0.66.15",
+    "@types/react-native-popup-dialog": "^0.16.4",
```
Comment (AnthonyHaykal) [inline package.json:45]: same as abov

File: src/excels/sorttingfilters/sorts/SortByDate.tsx (hunk 1 above)
Comment (mounir95) [reply src/excels/sorttingfilters/sorts/SortByDate.tsx]: noted, yes and done

File: package.json (hunk 3 above)
Comment (mounir95) [reply package.json:30]: yes i am using it to show the popUp, import Dialog, {DialogTitle, DialogContent} from 'react-native-popup-dialog';

Comment (AnthonyHaykal) [reply package.json:30]: but you didn't use it in this branch, it was from a previous branch?

File: package.json (hunk 4 above)
Comment (mounir95) [reply package.json:45]: regarding this i get an error when i remove it i get
Could not find a declaration file for module 'react-native-popup-dialog'.
Try `npm i --save-dev @types/react-native-popup-dialog`, so i shpuld use only this command without yarn add react-native-popup-dialog ?

**Pull request merged**